### PR TITLE
securenn

### PR DESCRIPTION
**Pull request by lwxxxxxxx**
《SecureNN 密态协议研究和优化》项目实现和初步优化代码

### CODE REVIEW

File: libspu/spu.proto [hunk 1]
```diff
@@ -115,6 +115,8 @@ enum ProtocolKind {
   // The famous [Cheetah](https://eprint.iacr.org/2022/207) protocol, a very
   // fast 2PC protocol.
   CHEETAH = 4;
+
+  SECURENN = 5;
```
Comment (anakinxc) [inline libspu/spu.proto:119]: Like other protocols above, add some comments about secrenn here.

File: libspu/mpc/factory.cc [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #include "libspu/mpc/cheetah/io.h"
 #include "libspu/mpc/cheetah/protocol.h"
 #include "libspu/mpc/ref2k/ref2k.h"
+#include "libspu/mpc/securenn/io.h"
+#include "libspu/mpc/securenn/protocol.h"
```
Comment (anakinxc) [inline libspu/mpc/factory.cc:26]: Seems missing cases in RegisterProtocol and CreateIO

File: libspu/mpc/securenn/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,141 @@
+# Copyright 2021 Ant Group Co., Ltd.
```
Comment (tpppppub) [inline libspu/mpc/securenn/BUILD.bazel:1]: Please update to 2023

File: libspu/mpc/securenn/protocol_test.cc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+RuntimeConfig makeConfig(FieldType field) {
+  RuntimeConfig conf;
+  conf.set_protocol(ProtocolKind::SECURENN);
+  conf.set_field(field);
+  return conf;
+}
+
+// std::once_flag init_server;
+// std::unique_ptr<brpc::Server> server;
```
Comment (rivertalk) [inline libspu/mpc/securenn/protocol_test.cc:33]: cleanup :P

**Review by tpppppub — APPROVED** [review]
LGTM

**Pull request merged**